zulipterminal/ui_tools/boxes.py

```diff
@@ -975,18 +975,16 @@ def soup2markup(cls, soup: Any, metadata: Dict[str, Any],
             elif tag in ('p', 'del'):
                 # PARAGRAPH, STRIKE-THROUGH
                 markup.extend(cls.soup2markup(element, metadata)[0])
-            elif (tag == 'span' and tag_attrs
-                  and 'emoji' in tag_classes):
+            elif tag == 'span' and 'emoji' in tag_classes:
                 # EMOJI
                 markup.append(('msg_emoji', tag_text))
-            elif (tag == 'span' and tag_attrs
-                  and ('katex-display' in tag_classes
-                       or 'katex' in tag_classes)):
+            elif tag == 'span' and (
+                 set(('katex-display', 'katex')) & set(tag_classes)):
                 # MATH TEXT
                 markup.append(tag_text)
-            elif (tag == 'span' and tag_attrs
-                  and ('user-group-mention' in tag_classes
-                       or 'user-mention' in tag_classes)):
+            elif tag == 'span' and (
+                 set(('user-group-mention', 'user-mention'))
+                 & set(tag_classes)):
```

Comment:
I do like how the set approach in the three cases keeps the strings together without requiring the explicit loops from before and like with the unrendered-div case.

Thoughts:
* simpler with `{...}` instead of `set` for the mentions?
* use a standard indent?
* minor: I'm used to us having operators on the following line, but this is OK

Reply:
wow, I thought we always had to use `set()`. I think now everything might fit in a single line :crossed_fingers:

Reply:
No, it doesn't fit in a single line :disappointed: but it looks much better now.